prov/efa/test/efa_unit_test_cq.c

```diff
@@ -149,20 +156,11 @@ void test_cq_read_bad_send_status(enum fi_ep_type ep_type)
 	ret = fi_cq_readerr(resource.cq, &cq_err_entry, 0);
 	assert_int_equal(ret, 1);
 
-	if (ep_type == FI_EP_DGRAM) {
-		assert_int_equal(cq_err_entry.err, FI_EIO);
-	} else {
+	if (ep_type != FI_EP_DGRAM) {
 		assert_int_equal(ep_type, FI_EP_RDM);
-		/* TODO:
-		 *
-		 * Our current behavior under such circumstance is to set cq_err_entry.err
-		 * to IBV_WC_GENERIC_ERR, which is not right.
-		 * cq_err_entry.err should be a libfabric error code.
-		 * We need to fix the behavior in provider, then update the assertion here.
-		 */
-		assert_int_equal(cq_err_entry.err, IBV_WC_GENERAL_ERR);
 	}
-	assert_int_equal(cq_err_entry.prov_errno, IBV_WC_GENERAL_ERR);
+	assert_int_equal(cq_err_entry.err, FI_EIO);
+	assert_in_range(cq_err_entry.prov_errno, FI_EFA_OK + 1, FI_EFA_IO_COMP_STATUS_MAX - 1);
```

Comment:
be definitive. use `assert_int_equal` to the error code you set earlier.

Also call `fi_cq_strerr(cq_err_entry.err, cq_err_entry.prov_error)` and make sure the string returned has correct information.

Reply:
Oops, forgot to address your second comment here before pushing changes again. Let me fix that.